proof/refine/ARM/TcbAcc_R.thy

```diff
@@ -1442,6 +1425,13 @@ lemma threadSet_valid_tcbs':
   apply (clarsimp simp: obj_at'_def projectKOs valid_tcbs'_def)
   done
 
+lemma asUser_valid_tcbs'[wp]:
+  "asUser t f \<lbrace>valid_tcbs'\<rbrace>"
+  apply (simp add: asUser_def split_def)
+  apply (wpsimp wp: threadSet_valid_tcbs' hoare_drop_imps
+              simp: valid_tcb'_def tcb_cte_cases_def)
```

Comment:
optional: can one-line this by adding the initial defs to wpsimp's simp

Reply:
I feel like this is almost another style question. This approach of having an explicit step unfolding the definitions separately to the 'actual proof' is very common.

Personally, I find it useful when repairing a broken proof to be able to so easily see the definition of the function, but this isn't enough of a benefit for me to care much. It would be nice however for us to make a conscious decision about what the preferred style is so that we can be consistent going forwards.

Reply:
Agreed, but if we are unfolding, I prefer `unfolding X Y by ...` to the double-simp.